src/librustc/ty/instance.rs

```diff
@@ -374,6 +375,29 @@ impl<'tcx> Instance<'tcx> {
     pub fn is_vtable_shim(&self) -> bool {
         if let InstanceDef::VtableShim(..) = self.def { true } else { false }
     }
+
+    pub fn polymorphize(&mut self, tcx: TyCtxt<'tcx>) {
```

Comment:
Also I think I would rename this to "`generalize`" or, well, have a bikeshed.

And maybe this would be better as a `self -> Self` method, instead of mutating.

Reply:
I've added a doc comment and made this function by-value in ac01dac.